modules/auth/proc-adding-gitlab-oauth-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,61 @@
+[id='proc-adding-gitlab-oauth-as-an-authentication-provider_{context}']
+= Enabling the GitLab OAuth authentication provider
+
+{product} includes a GitLab authentication provider that can authenticate users by using GitLab OAuth.
+
+.Procedure
+. To allow {product-short} to authenticate with Gitlab, create an OAuth Application in Gitlab.
++
+Go to https://gitlab.com/-/profile/applications[GitLab User settings > Applications], and click the *Add new application* button.
+
+*Name*:: Enter your application name, such as `{product-short}`.
+*Redirect URI*:: Enter the backend authentication URI set in {product-short}:
+For local development::: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+For a non-local deployment::: `pass:c,a,q[http://_<APP_FQDN>_/api/auth/gitlab/handler/frame]`
++
+NOTE: Due to a peculiarity with GitLab OAuth, ensure the URL has no trailing `/` after 'frame'.
+*Scopes*:: Select the following scopes from the list, and validate:
++
+// NOTE: including here verbatim copy from Gitlab screen.
+`read_user`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`read_repository`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`write_repository`::: Grants read/write access to repositories on private projects using Git-over-HTTP (not using the API).
+`openid`::: Grants permission to authenticate with GitLab using OpenID Connect.
+Also gives read-only access to the user's profile and group memberships.
+`profile`::: Grants read-only access to the user's profile data using OpenID Connect.
+`email`::: Grants read-only access to the user's primary email address using OpenID Connect.
++
+Save the *Application ID* and *Secret* for the next step.
+
+. Set up the Gitlab authentication provider in {product-short} configuration (`app-config.yaml`).
++
+Add the `gitlab` provider configuration to your `app-config.yaml` under the root `auth` configuration, and optionally, enable the `gitlab` provider for sign-in:
++
+.`app-config.yaml` fragment
+[source,yaml,subs="+quotes,+attributes"]
+----
+auth:
+  environment: development
+  providers:
+    gitlab:
+      development:
+        clientId: ${AUTH_GITLAB_CLIENT_ID} # <1>
+        clientSecret: ${AUTH_GITLAB_CLIENT_SECRET} # <2>
+        # audience: https://gitlab.company.com # <3>
+        # callbackUrl: https://_<APP_FQDN>_/api/auth/gitlab/handler/frame # <4>
+# signInPage: gitlab # <5>
+----
++
+<1> Enter the *Application ID* that you generated on GitLab, such as
```

Comment:
Should we guide users toward using Kubernets Secrets to store those values, as it is done in https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.1/html/getting_started_with_red_hat_developer_hub/ref-rhdh-supported-configs_rhdh-getting-started#configuring-github-authentication

Or is safe to assume that they will know better than adding secrets directly into the config?

It might be nice if we have some generic section about how to handle secrets and just refer to it in places like here.

Reply:
Indeed.

We can have one secret handling multiple key/values, right?